pkg/controller/registry/resolver/resolver.go

```diff
@@ -748,18 +753,73 @@ func DependencyPredicates(properties []*api.Property) ([]cache.Predicate, error)
 	return predicates, nil
 }
 
-func predicateForProperty(property *api.Property) (cache.Predicate, error) {
+func (pc *predicateConverter) predicateForProperty(property *api.Property) (cache.Predicate, error) {
 	if property == nil {
 		return nil, nil
 	}
-	p, ok := predicates[property.Type]
+
+	if property.Type == constraints.OLMConstraintType {
+		return pc.predicateForConstraintProperty(property.Value)
+	}
+
+	// Legacy behavior dictates that unknown properties are ignored.
```

Comment:
Could it be worth adding more context here? (Maybe point to the discussion in the EP). Just so the next dev understands what's happening and why.